### PR TITLE
[PULL REQUEST]Add translation/zh_CN.po

### PR DESCRIPTION
**Pull request by powertide**
This is my very first attempt for pull request.
I was not so sure about some of the translations, which I have marked.
So, check it out please.

### CODE REVIEW

**Comment by powertide** [comment]
What does this mean "All checks have failed"? @lordofhyphens 
Should I do anything about it?

**Comment by lordofhyphens** [comment]
It's not related to your changes.

File: translation/zh_CN.po [hunk 1]
```diff
@@ -0,0 +1,1938 @@
...
+
+#: xs/src/libslic3r/PrintConfig.cpp:20
+msgid "Hilbert Curve"
+msgstr "希尔伯特曲线式"
+
+#: xs/src/libslic3r/PrintConfig.cpp:21
+msgid "Archimedean Chords"
+msgstr "阿基米德和铉式"
+
+# Not so sure about how to put "Octagram Spiral"  in Chinese. Or maybe "Octagram螺旋“ is good.
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: This is up for debate, obviously. If someone else disagrees, they can put up a pull request.

File: translation/zh_CN.po [hunk 2]
```diff
@@ -0,0 +1,1938 @@
...
+"periodically wiped."
+msgstr ""
+"当挤出头处于非活跃状态时用到的温度差。启用一个全高度的环边，纯用于定期给喷嘴"
+"擦拭。"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1247 xs/src/libslic3r/PrintConfig.cpp:1256
+msgid "Start G-code"
+msgstr "起始G代码"
+
+# “before extruder has finished heating. If Slic3r detects M104, M109, M140 or M190 in your custom codes, such commands will not be prepended automatically so you're free to customize the order of heating commands and other custom actions. ”
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: This is for custom start g-code. If  you do not have M104, M109, M140, or M190 in the box then the text will be inserted after a M190 and M104 that is automatically placed by Slic3r. If the box contains any of those four gcodes the automatic insertion of M190 and M104 will be suppressed and this test will be at the beginning of the gcode.

File: translation/zh_CN.po [hunk 3]
```diff
@@ -0,0 +1,1938 @@
...
+"板位置进行补偿，如果你的终点挡板为零时，喷嘴离打印热床还有0.3毫米，将该值设"
+"为-0.3（或者调整你的终点挡板）。"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1573
+msgid "Z full steps/mm"
+msgstr "Z全部步骤数/毫米"
+
+# "you can calculate this by dividing the number of microsteps configured in your firmware by the microstepping amount (8, 16, 32). "
+# Lost me there.
+# What is "the number of microsteps" and "microstepping amount"?
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: Stepper drivers have a fixed multiplier (the common ones support either 16x or 32x).

Each stepper driver, however has a fixed number of full steps per revolution. This setting is to permit automatic adjustment of Z height to increase/decrease Z by the stepper driver's full step. The rest of the text is a way to calculate the number of mm for each full step.

File: translation/zh_CN.po [hunk 4]
```diff
@@ -0,0 +1,1938 @@
...
+"example, if your endstop zero actually leaves the nozzle 0.3mm far from the "
+"print bed, set this to -0.3 (or fix your endstop)."
+msgstr ""
+"在输出G代码中，所有Z坐标将在原基础上增大（或减少）该数值。用来对糟糕的Z终点挡"
+"板位置进行补偿，如果你的终点挡板为零时，喷嘴离打印热床还有0.3毫米，将该值设"
+"为-0.3（或者调整你的终点挡板）。"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1573
+msgid "Z full steps/mm"
+msgstr "Z全部步骤数/毫米"
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: Is it common to replace unit designators (mm, cm, g, etc) with Chinese characters?

Comment (powertide) [inline translation/zh_CN.po]: Yes, Chinese would understand both.
Both are fine.

**Comment by powertide** [comment]
Still,  4 translations unfinished...
Would you care to explain a little more about them please ? @lordofhyphens

File: translation/zh_CN.po [hunk 5]
```diff
@@ -0,0 +1,1926 @@
...
+
+#: xs/src/libslic3r/PrintConfig.cpp:1741
+msgid "Export the model as OBJ."
+msgstr "将模型以OBJ格式导出。"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1746
+msgid "Export POV"
+msgstr "导出POV"
+
+# POV-Ray?
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: It's a freeware raytracing program: http://www.povray.org/

File: translation/zh_CN.po [hunk 6]
```diff
@@ -0,0 +1,1926 @@
...
+
+#: xs/src/libslic3r/PrintConfig.cpp:1409
+msgid "Speed for printing support material."
+msgstr "打印支撑材料的速度"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1418
+msgid "Overhang threshold"
+msgstr "悬空阈值"
+
+# "% o“ format mistake?
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: No, this isn't a printf-formatted string. The % is literal.

File: translation/zh_CN.po [hunk 7]
```diff
@@ -0,0 +1,1926 @@
...
+"Only create support if it lies on a build plate. Don't create support on a "
+"print."
+msgstr "仅在热床上生成支撑，不在打印出来的物体上生成。"
+
+#: xs/src/libslic3r/PrintConfig.cpp:1294
+#, fuzzy
+msgid "Contact Z distance"
+msgstr "接触Z距离"
+
+# How come " Setting this to 0 will also prevent *speed* for the first object layer"?
```
Comment (lordofhyphens) [inline translation/zh_CN.po]: Bridging flow has its own special speed category. Setting support distance to 0 means that the first object layer over the interface will no longer be considered a bridge.

File: translation/zh_CN.po (hunk 6 above)
Comment (powertide) [inline translation/zh_CN.po]: Got it.
The font color confused me for a while.

**Comment by powertide** [comment]
Done here~

**Comment by lordofhyphens** [comment]
Thank you! There will be more text to work on later as we move more of the UI to C++.

**Comment by powertide** [comment]
You are welcome. 
I am more than glad to help contribute to this  community :-)

**Pull request merged**